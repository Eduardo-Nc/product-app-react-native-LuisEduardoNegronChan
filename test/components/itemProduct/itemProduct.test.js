import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react-native';
import ItemProduct from '@Components/Item/itemProduct';

const mockItem = {
  title: 'Producto de prueba',
  price: 199.99,
  category: 'Electrónica',
  image: 'https://example.com/product.jpg'
};

const mockFormattedPrice = (price) => `$${price.toFixed(2)}`;
const mockAction = jest.fn();

describe('Prueba del componente TxtInput', () => {
  it('Renderiza el título, precio formateado, categoría e imagen', () => {
    render(
      <ItemProduct
        item={mockItem}
        formattedPrice={mockFormattedPrice}
        action={mockAction}
        index={0}
        isLeftColumn={true}
      />
    );

    expect(screen.getByText(mockItem.category)).toBeTruthy();
    expect(screen.getByText(mockItem.title)).toBeTruthy();
    expect(screen.getByText(mockFormattedPrice(mockItem.price))).toBeTruthy();
  });

  it('Coincide con el snapshot visual', () => {
    const tree = render(
      <ItemProduct
        item={mockItem}
        formattedPrice={mockFormattedPrice}
        action={mockAction}
        index={0}
        isLeftColumn={false}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});