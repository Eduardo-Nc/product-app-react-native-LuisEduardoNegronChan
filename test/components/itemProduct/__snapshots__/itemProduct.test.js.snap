// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente TxtInput Coincide con el snapshot visual 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "#000000",
      "borderRadius": 10,
      "borderWidth": 0.3,
      "height": 230,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginLeft": "2%",
      "opacity": 1,
      "paddingHorizontal": 25,
      "width": "46%",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 12,
        "fontWeight": "600",
        "marginBottom": 5,
      }
    }
  >
    Electr√≥nica
  </Text>
  <Image
    source={
      {
        "uri": "https://example.com/product.jpg",
      }
    }
    style={
      {
        "height": 80,
        "overflow": "hidden",
        "resizeMode": "contain",
        "width": 80,
      }
    }
  />
  <Text
    adjustsFontSizeToFit={true}
    numberOfLines={3}
    style={
      {
        "color": "#000000",
        "fontSize": 14,
        "fontWeight": "400",
        "marginTop": 5,
        "textAlign": "left",
      }
    }
  >
    Producto de prueba
  </Text>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 16,
        "fontWeight": "500",
      }
    }
  >
    $199.99
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#282d36",
        "borderRadius": 100,
        "height": 20,
        "justifyContent": "center",
        "marginTop": 10,
        "overflow": "hidden",
        "width": 20,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../ProductApp/assets/icons/rightArrow.png",
        }
      }
      style={
        {
          "height": 8,
          "resizeMode": "contain",
          "tintColor": "#FFFFFF",
          "width": 8,
        }
      }
    />
  </View>
</View>
`;
